sklearn/preprocessing/_target_encoder.py

```diff
@@ -0,0 +1,165 @@
+import numpy as np
+
+from ..utils.validation import check_is_fitted
+from ..utils.validation import column_or_1d
+from ._encoders import _BaseEncoder
+
+
+class TargetRegressorEncoder(_BaseEncoder):
+    """Target Encoder for Regression Targets.
+
+    Each category is encoded based its effect on the target variable. The
+    encoding scheme takes a weighted average estimated by a multilevel
+    linear model.
+
+    Read more in the :ref:`User Guide <target_regressor_encoder>`.
+
+    Parameters
+    ----------
+    categories : 'auto' or a list of array-like, default='auto'
+        Categories (unique values) per feature:
+
+        - 'auto' : Determine categories automatically from the training data.
+        - list : `categories[i]` holds the categories expected in the ith
+          column. The passed categories should not mix strings and numeric
+          values within a single feature, and should be sorted in case of
+          numeric values.
+
+        The used categories can be found in the `categories_` attribute.
+
+    Attributes
+    ----------
+    cat_encodings_ : list of ndarray
+        The encoding corresponding to the categories in the `categories_`
+        attribute.
+
+    categories_ : list of ndarray
+        The categories of each feature determined during fitting
+        (in order of the features in X and corresponding with the output
+        of :meth:`transform`).
+
+    y_mean_ : float
+        The overall mean of the target.
+
+    See Also
+    --------
+    sklearn.preprocessing.OrdinalEncoder : Performs an ordinal (integer)
+      encoding of the categorical features.
+    sklearn.preprocessing.OneHotEncoder : Performs a one-hot encoding of
+      categorical features.
```

Comment:
I think we should avoid references in the docstrings because they lack context, and we sometimes end up with big lists of refs that are impractical because some of them are about super specific details.

It's better to have them in the UG so that we can explicitly refer to them as in "this is detailed in blahblah" or "this was introduced in ...".

In my recent UG-cleaning PRs, I've removed refs from docstrings.